In [1]:
import pandas as pd

# Read the transaction and purchase behavior datasets
transaction_data = pd.read_excel('QVI_transaction_data.xlsx')
purchase_behaviour = pd.read_csv('QVI_purchase_behaviour.csv')

# Merge datasets on LYLTY_CARD_NBR
merged_data = transaction_data.merge(purchase_behaviour, on='LYLTY_CARD_NBR', how='inner')

# Check high-level summary
print("Merged Dataset Info:")
print(merged_data.info())


Merged Dataset Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 264836 entries, 0 to 264835
Data columns (total 10 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   DATE              264836 non-null  int64  
 1   STORE_NBR         264836 non-null  int64  
 2   LYLTY_CARD_NBR    264836 non-null  int64  
 3   TXN_ID            264836 non-null  int64  
 4   PROD_NBR          264836 non-null  int64  
 5   PROD_NAME         264836 non-null  object 
 6   PROD_QTY          264836 non-null  int64  
 7   TOT_SALES         264836 non-null  float64
 8   LIFESTAGE         264836 non-null  object 
 9   PREMIUM_CUSTOMER  264836 non-null  object 
dtypes: float64(1), int64(6), object(3)
memory usage: 20.2+ MB
None


In [3]:
# Total sales by segment
sales_by_segment = merged_data.groupby(['LIFESTAGE', 'PREMIUM_CUSTOMER'])['TOT_SALES'].sum().reset_index()
sales_by_segment = sales_by_segment.sort_values(by='TOT_SALES', ascending=False)
print("\nTotal Sales by Segment:")
print(sales_by_segment)


Total Sales by Segment:
                 LIFESTAGE PREMIUM_CUSTOMER  TOT_SALES
6           OLDER FAMILIES           Budget  168363.25
19   YOUNG SINGLES/COUPLES       Mainstream  157621.60
13                RETIREES       Mainstream  155677.05
15          YOUNG FAMILIES           Budget  139345.85
9    OLDER SINGLES/COUPLES           Budget  136769.80
10   OLDER SINGLES/COUPLES       Mainstream  133393.80
11   OLDER SINGLES/COUPLES          Premium  132263.15
12                RETIREES           Budget  113147.80
7           OLDER FAMILIES       Mainstream  103445.55
14                RETIREES          Premium   97646.05
16          YOUNG FAMILIES       Mainstream   92788.75
1   MIDAGE SINGLES/COUPLES       Mainstream   90803.85
17          YOUNG FAMILIES          Premium   84025.50
8           OLDER FAMILIES          Premium   81958.40
18   YOUNG SINGLES/COUPLES           Budget   61141.60
2   MIDAGE SINGLES/COUPLES          Premium   58432.65
20   YOUNG SINGLES/COUPLES          Prem

[INSIGHTS]

Top Spending Segments (Total Sales):
Older Families (Budget): This segment spends the most ($168,363.25), indicating they are key customers.
Young Singles/Couples (Mainstream) and Retirees (Mainstream) also contribute significantly to total sales.
Budget and Mainstream segments dominate across different lifestages, showing value-conscious customers are driving sales.

In [5]:
# Average product quantity purchased by segment
avg_qty_by_segment = merged_data.groupby(['LIFESTAGE', 'PREMIUM_CUSTOMER'])['PROD_QTY'].mean().reset_index()
avg_qty_by_segment = avg_qty_by_segment.sort_values(by='PROD_QTY', ascending=False)
print("\nAverage Product Quantity by Segment:")
print(avg_qty_by_segment)


Average Product Quantity by Segment:
                 LIFESTAGE PREMIUM_CUSTOMER  PROD_QTY
8           OLDER FAMILIES          Premium  1.980969
7           OLDER FAMILIES       Mainstream  1.948610
6           OLDER FAMILIES           Budget  1.945812
15          YOUNG FAMILIES           Budget  1.940749
16          YOUNG FAMILIES       Mainstream  1.940342
17          YOUNG FAMILIES          Premium  1.937732
11   OLDER SINGLES/COUPLES          Premium  1.914273
9    OLDER SINGLES/COUPLES           Budget  1.913403
1   MIDAGE SINGLES/COUPLES       Mainstream  1.911656
10   OLDER SINGLES/COUPLES       Mainstream  1.910525
14                RETIREES          Premium  1.900122
12                RETIREES           Budget  1.892244
0   MIDAGE SINGLES/COUPLES           Budget  1.891633
2   MIDAGE SINGLES/COUPLES          Premium  1.889727
13                RETIREES       Mainstream  1.887543
5             NEW FAMILIES          Premium  1.860919
4             NEW FAMILIES       Mainstream 

[INSIGHTS]

Average Product Quantity Purchased by Segment:
Older Families and Young Families consistently purchase more products per transaction (close to 2 units per purchase).
Premium customers in most segments show slightly higher purchasing behavior compared to Budget and Mainstream.

In [7]:
# Analyze preferred pack sizes
# Extract pack size (assuming it’s embedded in `PROD_NAME` like 'Xg' or 'XXg')
merged_data['PACK_SIZE'] = merged_data['PROD_NAME'].str.extract(r'(\d+)[gG]')
pack_size_by_segment = merged_data.groupby(['LIFESTAGE', 'PREMIUM_CUSTOMER', 'PACK_SIZE'])['TOT_SALES'].sum().reset_index()
pack_size_by_segment = pack_size_by_segment.sort_values(by='TOT_SALES', ascending=False)
print("\nPack Size Preference by Segment:")
print(pack_size_by_segment.head(10))


Pack Size Preference by Segment:
                 LIFESTAGE PREMIUM_CUSTOMER PACK_SIZE  TOT_SALES
134         OLDER FAMILIES           Budget       175    42204.7
281               RETIREES       Mainstream       175    38242.7
407  YOUNG SINGLES/COUPLES       Mainstream       175    37967.9
323         YOUNG FAMILIES           Budget       175    35634.8
197  OLDER SINGLES/COUPLES           Budget       175    34497.0
239  OLDER SINGLES/COUPLES          Premium       175    33393.3
218  OLDER SINGLES/COUPLES       Mainstream       175    33041.6
260               RETIREES           Budget       175    28977.1
130         OLDER FAMILIES           Budget       150    27017.1
155         OLDER FAMILIES       Mainstream       175    25975.4


[INSIGHTS]

Pack Size Preferences:
175g packs are the clear favorite across all segments, especially for Older Families (Budget) and Retirees (Mainstream).
150g packs are also popular, but to a lesser extent.

In [9]:
# Analyze preferred brands
# Extract brand name (assuming it’s the first word in `PROD_NAME`)
merged_data['BRAND'] = merged_data['PROD_NAME'].str.split().str[0]
brand_by_segment = merged_data.groupby(['LIFESTAGE', 'PREMIUM_CUSTOMER', 'BRAND'])['TOT_SALES'].sum().reset_index()
brand_by_segment = brand_by_segment.sort_values(by='TOT_SALES', ascending=False)
print("\nTop Brands by Segment:")
print(brand_by_segment.head(10))


Top Brands by Segment:
                  LIFESTAGE PREMIUM_CUSTOMER   BRAND  TOT_SALES
563   YOUNG SINGLES/COUPLES       Mainstream  Kettle    35423.6
186          OLDER FAMILIES           Budget  Kettle    32058.0
389                RETIREES       Mainstream  Kettle    31652.4
273   OLDER SINGLES/COUPLES           Budget  Kettle    29066.4
331   OLDER SINGLES/COUPLES          Premium  Kettle    27943.4
302   OLDER SINGLES/COUPLES       Mainstream  Kettle    26852.8
447          YOUNG FAMILIES           Budget  Kettle    26369.6
360                RETIREES           Budget  Kettle    24340.0
418                RETIREES          Premium  Kettle    20922.4
41   MIDAGE SINGLES/COUPLES       Mainstream  Kettle    20231.8


[INSIGHTS]

Top Brands:
Kettle is the most preferred brand across all lifestages and customer tiers, particularly among Young Singles/Couples (Mainstream) and Older Families (Budget).
This highlights a strong brand affinity, making Kettle a key product to stock or promote.

Recommendations:

Marketing Strategy:
Focus on Budget and Mainstream customers, especially in segments like Older Families and Young Singles/Couples, as they dominate sales.
Develop targeted promotions for Kettle and emphasize 175g packs in campaigns.

Product Stocking:
Ensure adequate inventory of 175g packs in stores catering to Budget and Mainstream customers, particularly for popular segments.

Loyalty Programs:
Introduce tailored loyalty programs to engage high-spending segments (e.g., Budget Older Families).
Offer discounts on bulk purchases to encourage higher product quantities per transaction.